homeassistant/components/zwave/__init__.py

```diff
@@ -376,6 +376,26 @@ def _on_timeout(sec):
         hass.add_job(check_has_unique_id, entity, _on_ready, _on_timeout,
                      hass.loop)
 
+    def node_removed(node):
+        node_id = node.node_id
+        node_key = 'node-{}'.format(node_id)
+        _LOGGER.info("Node Removed: %s",
+                     hass.data[DATA_DEVICES][node_key])
+        for key in list(hass.data[DATA_DEVICES]):
+            if not key.startswith('{}-'.format(node_id)):
+                continue
+
+            entity = hass.data[DATA_DEVICES][key]
+            _LOGGER.info('Removing Entity - value: %s - entity_id: %s',
+                         key, entity.entity_id)
+            asyncio.run_coroutine_threadsafe(entity.node_removed(),
```

Comment:
Is this needed? Why can't this function be `async` and then you can just await this ?

Reply:
Since this is run by the dispatcher, async method is possible.

Reply:
I tried to await it instead of using run_coroutine_threadsafe but never could get it to work. I forget the exact errors, I think it was something about having an async function within a non-async context. I'll mess around with it more and see if I can get it working without `run_coroutine_threadsafe` (or atleast get the error message that it's giving)

Reply:
This is pydispatcher, which doesn’t support async.  You’ll have to schedule it, could you use hass.add_job?